src/utils/cache.rs

```diff
@@ -96,22 +97,38 @@ pub trait CacheExt: Cache {
         }
     }
 
-    fn request_cached(
+    // TODO: hate this name
+    fn request_cached_advanced(
         &self,
         client: &reqwest::blocking::Client,
         url: &str,
-        timeout: usize,
-        error_timeout: usize,
+        cache_duration: usize,
+        error_cache_duration: usize,
+        cache_all_errors: bool,
+        request_timeout: u64,
     ) -> ApiResult<String> {
         let cache_key = format!("{}_{}", CACHE_REQS_PREFIX, &url);
         match self.fetch(&cache_key) {
             Some(cached) => CachedWithCode::split(&cached).to_result(),
             None => {
-                let response = client.get(url).send()?;
+                let mut request = client.get(url);
+                if request_timeout > 0 {
+                    request = request.timeout(Duration::from_millis(request_timeout));
+                }
+                let client_response = request.send();
+                if cache_all_errors && client_response.is_err() {
+                    self.create(
+                        &cache_key,
+                        CachedWithCode::join(500, "").as_str(),
```

Comment:
Not sure what the empty string represents here. Is there something more meaningful we could put here for easier debugging in the future?

Reply:
Probably the err object could be serialized